rust/1.58.0/Dockerfile

```diff
@@ -1,80 +0,0 @@
-ARG RUST_VERSION=1.58.0
```

Comment:
Could you please elaborate why this Dockerfile for 1.58.0 is removed?

Reply:
This is a build argument, both files were almost identical, with the same build arguments, but their default values being different: so I've collapsed files to one and specified on extra build arg in the build scripts.